frontend/rust-lib/flowy-database2/src/entities/type_option_entities/date_entities.rs

```diff
@@ -32,6 +33,12 @@ pub struct DateCellDataPB {
 
   #[pb(index = 8)]
   pub is_range: bool,
+
+  #[pb(index = 9)]
+  pub reminder_id: String,
+
+  #[pb(index = 10)]
+  pub reminder_option: ReminderOptionPB,
```

Comment:
The cell data only needs the reminder id. The option should be stored inside the reminder.

Reply:
I'm not so sure, because the Reminder doesn't care at all about the option.

It's more of a pseudo-entity that is added only to provide context to Clients when adding/editing a Reminder on a Date.

Reply:
I think it would be nice to have this option with every reminder, Notion has this as well. This is a use case that I have in mind.

I have a meeting doc with a list of participants, agenda, meeting, notes, etc. I set the time for Monday Jan 1, at 9:00 and set up a reminder to remind me 10 minutes before.

Reply:
Yes, and again my point is, you're adding a Reminder to a Date, `Mon, 1st Jan 9:00`.

I'm not sure why the Reminder should care what option it is, it only cares about the Date of it's trigger? Changing the Option is a "Client"/UX behavior.

Reply:
We can even go as far to completely remove `ReminderOptionPB`, and instead infer it on the frontend by taking the Reminder Date and subtracting from the Event Date, and then matching the difference.

```
final reminderOption = _reminderOptionFromDates(event.date, reminder.scheduledAt);

ReminderOption _reminderOptionFromDates(
  DateTime eventDate,
  DateTime reminderDate,
) {
  final diff = eventDate.difference(reminderDate);
  
  return switch (diff.inMinutes) {
    Duration(minutes: 0) => ReminderOption.atTimeOfEvent,
    Duration(minutes: 10) => ReminderOption.tenMinsBefore,
    Duration(minutes: 15) => ReminderOption.fifteenMinsBefore,
    ...
    _ => ReminderOption.custom,
  };
}
```
If you want, I can completely remove ReminderOptionPB in favor of the above?

Reply:
Ahh I get it now, sorry I got confused.

Yes, think it would be better to infer the option from reminder time. That way we reduce potential bugs if one side didn't get updated, or if we want to be able to reschedule from the notification hub in the future.